src/test/test_framework/util.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TEST_TEST_FRAMEWORK_UTIL_H
+#define BITCOIN_TEST_TEST_FRAMEWORK_UTIL_H
+
+#include <string>
+#include <vector>
+
+void assert_debug_log_redirect();
+void assert_debug_log_helper(const std::vector<std::string>& messages);
+
+#define ASSERT_DEBUG_LOG(vec_messages, code) [&] { \
```

Comment:
Why place in a lambda?

Reply:
To prevent `continue` and `break` without a loop or switch statement in the body.

Reply:
Got it, thanks.